src/algorithms/all_neighbors.c

```diff
@@ -17,12 +17,15 @@ static void _AllNeighborsCtx_CollectNeighbors
 	ctx->current_level++;
 	GxB_MatrixTupleIter *iter;
 	if(ctx->current_level == array_len(ctx->levels)) {
-		GxB_MatrixTupleIter_new(&iter, ctx->M);
-		array_append(ctx->levels, iter);
-	} else {
-		iter = ctx->levels[ctx->current_level];
+		ASSERT(ctx->M);
```

Comment:
Move this line and `GxB_set(ctx->M, GxB_SPARSITY_CONTROL, GxB_SPARSE) ;` to:
both `AllNeighborsCtx_Reset` and `AllNeighborsCtx_New`